tools/lts-release.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+
+#
+# Copyright (c) 2024 Zededa, Inc.
+# SPDX-License-Identifier: Apache-2.0
+#
+
+git_branch=$(git rev-parse --abbrev-ref HEAD | tr / _)
+
+# Get the latest tag from the specified branch
+latest_tag=$(git describe --tags --abbrev=0 "$git_branch")
+
+
+# Check if the input is a valid release branch in the format X.Y (e.g., 1.2)
+check_release_branch() {
+    if ! echo "${git_branch}" | grep -Eq "^[0-9]+\.[0-9]+(-[a-zA-Z]+)?$"; then
+        echo "ERROR: must be on a release branch X.Y"
+        exit 1
+    fi
+}
+
+check_tag() {
+    if ! echo "${latest_tag}" | grep -Eq "^[0-9]+\.[0-9]+\.[0-9]+(-[a-zA-Z0-9]+)?$"; then
+        echo "ERROR: can't find previous release's tag X.Y.Z"
+        exit 1
+    fi
+}
+
+# Define a function to increment the patch version
+increment_patch() {
+    local version=$1
+    local major minor patch
+    IFS='.' read -r major minor patch <<< "$version"
+    patch=$((patch + 1))
+    echo "$major.$minor.$patch"
+}
+
+# Check the suffix of the latest tag
+if [[ "$latest_tag" =~ -lts$ ]]; then
+    # If the tag ends with -lts, increment the patch version
+    base_version="${latest_tag%-lts}"  # Remove the -lts suffix
+    new_version=$(increment_patch "$base_version")
+    new_tag="${new_version}-lts"
+    echo "Incrementing patch version: $new_tag"
+elif [[ "$latest_tag" =~ -rc[0-9]+$ ]]; then
+    # If the tag ends with -rc, remove the rc suffix and check the patch version
+    base_version="${latest_tag%-rc*}"  # Remove the rc suffix
+    IFS='.' read -r major minor patch <<< "$base_version"
+    if [[ "$patch" == "0" ]]; then
+        # If the patch version is 0, create the first LTS release with the same base version
+        new_tag="${base_version}-lts"
+        echo "Creating the first LTS release: $new_tag"
+    else
+        # If the patch version is non-zero, create the LTS tag without incrementing the patch
+        new_tag="${base_version}-lts"
+        echo "Creating the LTS release: $new_tag"
+    fi
+else
+    # If the tag does not end with -lts or -rc, check if patch is 0 and create LTS
+    IFS='.' read -r major minor patch <<< "$latest_tag"
+    if [[ "$patch" == "0" ]]; then
+        # If the patch version is 0, create the first LTS release from the base version
+        new_tag="${latest_tag}-lts"
+        echo "Creating the first LTS release: $new_tag"
+    else
+        # If the patch version is non-zero, create an LTS tag by appending -lts
+        new_tag="${latest_tag}-lts"
+        echo "The latest tag does not end with -lts or -rc. Creating LTS tag: $new_tag"
+    fi
+fi
+
+# Tag the new version
+if ! git tag -a -m "Release ${new_tag}" "${new_tag}"; then
+    echo "Error: Failed to create a new tag."
+    exit 1
+fi
+
+echo "Tagged new version: $new_tag. Please push the tag to the remote repository"
+
+echo "git push origin $new_tag"
```

Comment:
Here and in the description you say to push to origin, but to date we have been doing all of these pushes to the release branch. Do you intend to change that?

Reply:
From what I understand, the previous patch command creates a tag from the currently checked-out branch using the `make` command `@echo "  git push origin $(REPO_BRANCH) $(PATCH_TAG)"`. This pushes `$(REPO_BRANCH)` to the origin, and since the branch already exists, we don't need to specify it explicitly. The `new_tag` will reference the branch based on the commit SHA.

I hope I have this right! :)

Reply:
FWIW Our make patch doesn't work that way. It outputs this:
nordmark@bobo:~/lf-edge/11.0-stable/eve$ make patch
Using kernel from KERNEL_TAG=docker.io/lfedge/eve-kernel:eve-kernel-amd64-v6.1.38-generic-820be3b1c356-gcc
Done tagging 11.0.8-lts patch release. Check the branch with git log and then run
  git push origin 11.0-stable 11.0.8-lts

And I've been blindly following that instruction. Thus it might work to push differently but I suggest we copy what make patch currently has unless someone who is an expert can tell us that it doesn't matter.

Reply:
For all the tags I've created, I'll just remove the `11.0-stable`, since that's the branch we're pushing to origin along with the `11.0.8-lts` tag :). The branch is already synced with the remote in this case. I tested the same command in the sample repo, and it only pushed the newly created tag. However, to maintain consistency with the previous process, I'll add the branch to both scripts at the end. It won't do us any harm :)